src/ir/possible-contents.cpp

```diff
@@ -900,21 +900,23 @@ struct InfoCollector
     if (curr->type == Type::unreachable) {
       return;
     }
-    auto heapType = curr->type.getHeapType();
-    switch (curr->op) {
-      case NewData: {
-        Type elemType = heapType.getArray().element.type;
-        addRoot(DataLocation{heapType, 0},
-                PossibleContents::fromType(elemType));
-        return;
-      }
-      case NewElem: {
-        Type segType = getModule()->elementSegments[curr->segment]->type;
-        addRoot(DataLocation{heapType, 0}, PossibleContents::fromType(segType));
-        return;
-      }
-    }
-    WASM_UNREACHABLE("unexpected op");
+    addRoot(curr, PossibleContents::exactType(curr->type));
+    // auto heapType = curr->type.getHeapType();
+    // switch (curr->op) {
+    //   case NewData: {
+    //     Type elemType = heapType.getArray().element.type;
+    //     addRoot(DataLocation{heapType, 0},
+    //             PossibleContents::fromType(elemType));
+    //     return;
+    //   }
+    //   case NewElem: {
+    //     Type segType = getModule()->elementSegments[curr->segment]->type;
+    //     addRoot(DataLocation{heapType, 0},
+    //             PossibleContents::fromType(segType));
+    //     return;
+    //   }
+    // }
+    // WASM_UNREACHABLE("unexpected op");
```

Comment:
@kripken, none of this code seems to affect the tests. Guidance on how to test it or whether to remove it would be appreciated.

Reply:
A module with just `array.new*` and `array.get` should work. Without the addRoot we'd have nothing for the `array.get` to read so it would be unreachable.